compaction/time_window_compaction_strategy.cc

```diff
@@ -26,53 +26,104 @@ time_window_compaction_strategy_state& time_window_compaction_strategy::get_stat
+const std::unordered_map<sstring, std::chrono::seconds> time_window_compaction_strategy_options::valid_window_units = {
+    { "MINUTES", 60s }, { "HOURS", 3600s }, { "DAYS", 86400s }
+};
+
+const std::unordered_map<sstring, time_window_compaction_strategy_options::timestamp_resolutions> time_window_compaction_strategy_options::valid_timestamp_resolutions = {
+    { "MICROSECONDS", timestamp_resolutions::microsecond },
+    { "MILLISECONDS", timestamp_resolutions::millisecond },
+};
+
+std::chrono::seconds time_window_compaction_strategy_options::validate_compaction_window_unit(const std::map<sstring, sstring>& options) {
     std::chrono::seconds window_unit = DEFAULT_COMPACTION_WINDOW_UNIT;
-    int window_size = DEFAULT_COMPACTION_WINDOW_SIZE;
 
-    auto it = options.find(COMPACTION_WINDOW_UNIT_KEY);
-    if (it != options.end()) {
-        auto valid_window_units_it = valid_window_units.find(it->second);
+    auto tmp_value = compaction_strategy_impl::get_value(options, COMPACTION_WINDOW_UNIT_KEY);
+    if (tmp_value) {
+        auto valid_window_units_it = valid_window_units.find(tmp_value.value());
         if (valid_window_units_it == valid_window_units.end()) {
-            throw exceptions::syntax_exception(sstring("Invalid window unit ") + it->second + " for " + COMPACTION_WINDOW_UNIT_KEY);
+            throw exceptions::syntax_exception(format("Invalid window unit {} for {}", tmp_value.value(), COMPACTION_WINDOW_UNIT_KEY));
         }
         window_unit = valid_window_units_it->second;
     }
 
-    it = options.find(COMPACTION_WINDOW_SIZE_KEY);
-    if (it != options.end()) {
-        try {
-            window_size = std::stoi(it->second);
-        } catch (const std::exception& e) {
-            throw exceptions::syntax_exception(sstring("Invalid integer value ") + it->second + " for " + COMPACTION_WINDOW_SIZE_KEY);
-        }
-    }
+    return window_unit;
+}
+
+std::chrono::seconds time_window_compaction_strategy_options::validate_compaction_window_unit(const std::map<sstring, sstring>& options, std::map<sstring, sstring>& unchecked_options) {
+    auto window_unit = validate_compaction_window_unit(options);
+    unchecked_options.erase(COMPACTION_WINDOW_UNIT_KEY);
+    return window_unit;
+}
+
+int time_window_compaction_strategy_options::validate_compaction_window_size(const std::map<sstring, sstring>& options) {
+    auto tmp_value = compaction_strategy_impl::get_value(options, COMPACTION_WINDOW_SIZE_KEY);
+    int window_size = cql3::statements::property_definitions::to_long(COMPACTION_WINDOW_SIZE_KEY, tmp_value, DEFAULT_COMPACTION_WINDOW_SIZE);
 
     if (window_size <= 0) {
-        throw exceptions::configuration_exception(fmt::format("{} must be greater than 1 for compaction_window_size", window_size));
+        throw exceptions::configuration_exception(format("{} must be greater than 1 for compaction_window_size", window_size));
```

Comment:
why `fmt::format` -> `format`?
We should prefer the former over the latter.
Also, where seastar is required, elsewhere we changed those occurrences to `::format`.